tests/WebdriverClassicConfig.php

```diff
@@ -15,22 +21,62 @@ public static function getInstance(): self
 
     public function createDriver(): DriverInterface
     {
-        return new WebdriverClassicDriver();
+        $browser = getenv('WEB_FIXTURES_BROWSER') ?: WebdriverClassicDriver::DEFAULT_BROWSER;
+        $seleniumHost = $_SERVER['DRIVER_URL'];
+
+        return new WebdriverClassicDriver($browser, [], $seleniumHost);
     }
 
-    public function skipMessage($testCase, $test): ?string
+    public function mapRemoteFilePath($file): string
     {
-        /** @phpstan-ignore-next-line */
-        if (true) {
-            return 'TODO: implement the initial driver';
+        if (!isset($_SERVER['TEST_MACHINE_BASE_PATH'])) {
+            $_SERVER['TEST_MACHINE_BASE_PATH'] = realpath(
+                    dirname(__DIR__) . '/vendor/mink/driver-testsuite/web-fixtures'
+                ) . DIRECTORY_SEPARATOR;
         }
 
-        /** @phpstan-ignore-next-line */
-        return parent::skipMessage($testCase, $test);
+        return parent::mapRemoteFilePath($file);
+    }
+
+    public function skipMessage($testCase, $test): ?string
+    {
+        switch (true) {
+            case $testCase === WindowTest::class && $test === 'testWindowMaximize' && $this->isXvfb():
+                return 'Maximizing the window does not work when running the browser in Xvfb.';
+
+            case $testCase === BasicAuthTest::class:
+                return 'Basic auth is not supported.';
+
+            case $testCase === HeaderTest::class:
+                return 'Headers are not supported.';
+
+            case $testCase === StatusCodeTest::class:
+                return 'Checking status code is not supported.';
+
+            case $testCase === EventsTest::class && $test === 'testKeyboardEvents' && $this->isOldChrome():
+                return 'Old Chrome does not allow triggering events.';
+
+            default:
+                return parent::skipMessage($testCase, $test);
+        }
     }
 
+    /**
+     * {@inheritdoc}
+     */
     protected function supportsCss(): bool
     {
         return true;
     }
+
+    private function isXvfb(): bool
```

Comment:
misleading method name. why you need it?

Reply:
> misleading method name
It's not exactly misleading - it's what we are actually intending there by checking the github action env var.

That check was originally implemented in MinkSelenium2Driver:
https://github.com/minkphp/MinkSelenium2Driver/blob/master/tests/Selenium2Config.php#L38

After trial and error, I found what it was trying to do, hence the name I gave it. Maybe we can find a better name though.

Reply:
Also, it gets its name from here: https://github.com/minkphp/webdriver-classic-driver/pull/1/files/d385eed7f7ec955f3706d765003a5907beb85509#diff-db0e666a5299190f519b12062eb766dccd38dc00a2c6813ec997ed4e6e29b186R45